ui/assets/js/hound.js

```diff
@@ -373,6 +374,23 @@ var SearchBar = React.createClass({
   filesGotFocus: function(event) {
     this.showAdvanced();
   },
+  excludeFilesGotKeydown: function(event) {
+    switch (event.keyCode) {
+    case 38:
+      // if advanced is empty, close it up.
+      if (this.refs.excludeFiles.getDOMNode().value.trim() === '') {
```

Comment:
I think this check made sense when there was only one text field, but now that there's two, you'll likely have to check that both this text field and the `this.refs.files` text field are empty or you'll end up closing the advanced view when it's still in use.

Reply:
Okay, I addressed that in the latest commit.